pkg/epp/plugins/plugins.go

```diff
@@ -50,3 +53,18 @@ type HandlePlugins interface {
 	// GetAllPluginsWithNames returns all of the known plugins with their names
 	GetAllPluginsWithNames() map[string]Plugin
 }
+
+// PluginByType retrieves the specified plugin by name and verifies its type
+func PluginByType[P Plugin](handle Handle, name string) (P, error) {
```

Comment:
looks good, can we pass plugins instead of handle as arg?

Reply:
Done